packages/@lwc/ssr-compiler/src/optimize-imports.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { produce } from 'immer';
+import { traverse } from 'estree-toolkit';
+import { Visitors } from './transmogrify';
+import type { Program as EsProgram } from 'estree';
+
+const visitors: Visitors = {
+    $: {
+        scope: true,
+    },
+    ImportDeclaration(path) {
+        const { node, scope } = path;
+        if (!node || !scope) {
+            return;
+        }
+        if (node.source.type === 'Literal' && node.source.value === '@lwc/ssr-runtime') {
+            node.specifiers = node.specifiers.filter(
+                (specifier) =>
+                    // There shouldn't be any default imports anyway
+                    specifier.type === 'ImportSpecifier' &&
+                    // If there are references, then the import is used
```

Comment:
```suggestion
                    // If there are references, then the import is used. Note that an import will _always_
                    // have a binding (of type "module"), but it may not have references.
```